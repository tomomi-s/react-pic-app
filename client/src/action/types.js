export const SET_ALERT = 'SET_ALERT';
export const REMOVE_ALERT = 'REMOVE_ALERT';

export const REGISTER_SUCCESS ='REGISTER_SUCCESS';
export const REGISTER_FAIL ='REGISTER_FAIL';
export const USER_LOADED ='USER_LOADED';
export const AUTH_ERROR ='AUTH_ERROR';
export const LOGIN_SUCCESS ='LOGIN_SUCCESS';
export const LOGIN_FAIL ='LOGIN_FAIL';
export const LOGOUT ='LOGOUT';
export const CLEAR_ERRORS ='CLEAR_ERRORS';

export const GET_POSTS = 'GET_POSTS';
export const ADD_POST = 'ADD_POST';
export const DELETE_POST = 'DELETE_POST';
export const UPDATE_POST = 'UPDATE_POST';
export const SET_CURRENT_POST = 'SET_CURRENT_POST';
export const POST_ERROR = 'POST_ERROR';
export const CLEAR_POST_ERRORS = 'CLEAR_POST_ERRORS';
export const SET_LOADING = 'SET_LOADING';


