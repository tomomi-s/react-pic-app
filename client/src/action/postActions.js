import axios from 'axios';

import { 
	GET_POSTS, 
	ADD_POST, 
	DELETE_POST, 
	UPDATE_POST,
	SET_CURRENT_POST,
	POST_ERROR,
	CLEAR_POST_ERRORS,
	SET_LOADING, } from './types';

//Get posts
export const getPosts = () => async dispatch => {
	try{
		setLoading();

		const res = await axios.get('/api/posts');
		dispatch({
			type: GET_POSTS,
			payload: res.data
		})
	}catch(err){
		dispatch({
			type: POST_ERROR, 
			payload: err.response.data.msg
		})
	}
}

//Get posts
export const getPost = (id) => async dispatch => {
	try{
		setLoading();

		const res = await axios.get(`/api/posts/${id}`);
		dispatch({
			type: SET_CURRENT_POST,
			payload: res.data
		})
	}catch(err){
		dispatch({
			type: POST_ERROR, 
			payload: err.response.data.msg
		})
	}
}

//Add new post
export const addPost = (post) => async dispatch =>{
	const config = {
			headers : {
				'Content-Type': 'multipart/form-data; boundary=<somestring>'
			}
	}

	try{
		setLoading();
		const res = await axios.post('/api/posts', post, config);

		dispatch({
			type: ADD_POST,
			payload: res.data
		})

	}catch(err){
		dispatch({
			type: POST_ERROR,
			payload: err.response.data.msg
		})
	}
}
//Delete post
export const deletePost = (id) => async dispatch => {
	try{
		setLoading();

		await axios.delete(`/api/posts/${id}`)

		dispatch({
			type: DELETE_POST,
			payload: id
		})

	}catch(err){
		dispatch({
			type: POST_ERROR,
			payload: err.response.data.msg
		})
	}
		
	
}
//Update post
export const updatePost = (post) => async dispatch => {
	console.log(post)
	const config = {
			headers : {
				'Content-Type': 'application/json'
			}
	}
	try{
		setLoading();

		const res = await axios.put(`/api/posts/${post._id}`, post, config);

		dispatch({
			type: UPDATE_POST,
			payload: res.data
		})

	}catch(err){
		dispatch({
			type: POST_ERROR,
			payload: err.response.data.msg
		})
	}
		
	
}

//Set current log
export const setCurrentPost = post => {
	return {
		type: SET_CURRENT_POST,
		payload: post
	}
}

// Set loading to true
export const setLoading = () => {
	return {
		type: SET_LOADING
	}
}

//Clear Errors
export const clearPostErrors = () => {return({type: CLEAR_POST_ERRORS})}