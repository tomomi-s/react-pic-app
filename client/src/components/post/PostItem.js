import React from 'react';
import { connect } from 'react-redux';
import { setCurrentPost } from '../../action/postActions';
import PropTypes from 'prop-types';

export const PostItem = ({post, setCurrentPost}) => {
	let tool = '';
	(post.tool === 1) ? tool = 'Adobe Lightroom'
	: (post.tool === 2) ? tool = 'VSCO'
	: tool = 'Snapseed';

	return (
		<div className="row" style={{ marginTop: 10 }}>


			<div className="card">
			    <div className="card-image waves-effect waves-block waves-light">
			      <img className="col s12 m6" style={{ padding: 0}} src={`/${post.photo_before['path']}`} alt=""/>
			      <img className="col s12 m6" style={{ padding: 0}} src={`/${post.photo_after['path']}`} alt=""/>
			    </div>
			    <div className="card-content">
			      <span className="card-title activator grey-text text-darken-4">{tool}<i className="material-icons right">more_vert</i></span>
			    </div>
			    <div className="card-reveal">
			      <span className="card-title grey-text text-darken-4">Recipe<i className="material-icons right">close</i></span>
			      <p>{post.process}</p>
			      <div className="fixed-action-btn">
					<a style={{ marginRight: 5}} className="btn-floating btn-large waves-effect waves-light teal lighten-1"  href={`/post/${post._id}`}><i className="material-icons">edit</i></a>
					<a href="#delete-post-modal" onClick={()=> setCurrentPost(post)} className="btn-floating btn-large waves-effect waves-light red lighten-2 modal-trigger"><i className="material-icons">delete</i></a>
				  </div>
			    </div>
			 </div>
			
		</div>
	)
}

PostItem.propTypes = {
	post: PropTypes.object,
	setCurrentPost: PropTypes.func.isRequired,
}
export default connect(null, { setCurrentPost })(PostItem)
