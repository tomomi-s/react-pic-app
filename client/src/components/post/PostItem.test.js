import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import { PostItem } from './PostItem';
import { setCurrentPost } from '../../action/postActions';

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

Enzyme.configure({ adapter: new Adapter() });

const post = {
	photo_before: {
		path: "/files/photo_before.jpg"
	},
	photo_after: {
		path: "/files/photo_after.jpg"
	},
	tool: 2,
	process: "Filter: C1"
}



it('render', ()=>{
	const wrapper = render(<PostItem post={post} setCurrentPost={setCurrentPost} />);
 	expect(wrapper).toMatchSnapshot();	
})

it('fire setCurrentPost function', ()=>{
	const mock = jest.fn();
	const wrapper = shallow(<PostItem post={post} setCurrentPost={mock} />);
	wrapper.find('.modal-trigger').simulate('click');
	expect(mock).toHaveBeenCalledTimes(1);
})