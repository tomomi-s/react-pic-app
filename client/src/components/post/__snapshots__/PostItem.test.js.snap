// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  class="row"
  style="margin-top:10px"
>
  <div
    class="card"
  >
    <div
      class="card-image waves-effect waves-block waves-light"
    >
      <img
        alt=""
        class="col s12 m6"
        src="//files/photo_before.jpg"
        style="padding:0"
      />
      <img
        alt=""
        class="col s12 m6"
        src="//files/photo_after.jpg"
        style="padding:0"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title activator grey-text text-darken-4"
      >
        VSCO
        <i
          class="material-icons right"
        >
          more_vert
        </i>
      </span>
    </div>
    <div
      class="card-reveal"
    >
      <span
        class="card-title grey-text text-darken-4"
      >
        Recipe
        <i
          class="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        Filter: C1
      </p>
      <div
        class="fixed-action-btn"
      >
        <a
          class="btn-floating btn-large waves-effect waves-light teal lighten-1"
          href="/post/undefined"
          style="margin-right:5px"
        >
          <i
            class="material-icons"
          >
            edit
          </i>
        </a>
        <a
          class="btn-floating btn-large waves-effect waves-light red lighten-2 modal-trigger"
          href="#delete-post-modal"
        >
          <i
            class="material-icons"
          >
            delete
          </i>
        </a>
      </div>
    </div>
  </div>
</div>
`;
